src/activities/bargame/Bargame.qml

```diff
@@ -0,0 +1,395 @@
+
+            // Hidden Answer Balls
+            Item {
+                id: answerBalls
+                // All green balls placement
+                Grid {
+                    id: answerBallsPlacement
+                    x: boxes.x
+                    y: boxes.y
+                    columns: Activity.boardSize[items.mode - 1]
+                    rows: 1
+                    Repeater {
+                        model: answerBallsPlacement.columns
+                        Image {
+                            source: Activity.url + "green_ball.svg"
+                            height: rootWindow.height / (9 + (items.mode - 1) * Activity.elementSizeFactor[items.mode - 1])
+                            width: rootWindow.width / (15 + (items.mode - 1) * Activity.elementSizeFactor[items.mode - 1])
+                            opacity: 0.0
+                        }
+                    }
+                }
+            }
+
+            // Masks
+            Grid {
+                id: masks
+                x: boxes.x
+                y: boxes.y
+                rows: 1
+                columns: Activity.boardSize[items.mode-1]
+                Repeater {
+                    id: startMask
+                    model: masks.columns
+                    Image {
+                        id: greenMask
+                        source: Activity.url + ((index == boxes.columns - 1) ? "mask_last.svg" : "mask.svg")
+                        height: rootWindow.height / (9 + (items.mode - 1) * Activity.elementSizeFactor[items.mode - 1])
+                        width: rootWindow.width / (15 + (items.mode - 1) * Activity.elementSizeFactor[items.mode - 1])
+                    }
+                }
+            }
+        }
+
+        // OK BUTTON
+        Image {
+            id: playLabel
+            x: rootWindow.width / 1.2
+            y: rootWindow.height / 2.4
+            width: rootWindow.width / 20
+            height: width
+            source: Activity.url + "bar_ok.svg"
+            MouseArea {
+                id: okArea
+                anchors.fill: parent
+                hoverEnabled: true
+                onClicked: {
+                    tuxArea.hoverEnabled = false;
+                    tuxArea.enabled = false;
+                    var value = Activity.numberOfBalls
+                    numberLabel.text = Activity.numberOfBalls = Activity.numberBalls[items.mode - 1][0];
+                    Activity.play(1, value);
+                }
+            }
+            states: State {
+                name: "mouseHover"
+                when: okArea.containsMouse
+                PropertyChanges {
+                    target: playLabel
+                    scale: 1.2
+                }
+            }
+        }
+
+        // Number of balls to be placed
+        Image {
+            id: ballNumberPlate
+            x: rootWindow.width / 1.2
+            y: rootWindow.height / 1.2
+            source: Activity.url + "enumerate_answer.svg"
```

Comment:
Maybe we can use only one button for it and don't change the source, I don't see the interest of having it green at first then yellow. It's never resetted to green anyway